### PR TITLE
SONOS Favorite support.

### PR DESCRIPTION
**Pull request by digitaldan**
Signed-off-by: Dan Cunningham dan@digitaldan.com

### CODE REVIEW

**Comment by digitaldan** [comment]
@kgoderis I finally had a chance to test this, so far so good, let me know your thoughts, thanks.

**Comment by kgoderis** [comment]
@digitaldan did you test this on the latest snapshot? (fixed a race condition in the meantime)

**Comment by digitaldan** [comment]
@kgoderis I did, it should be up to date with master as of yesterday or so

**Comment by digitaldan** [comment]
@kgoderis Have you had a chance to make sure I didn't break anything ;-)

**Comment by kgoderis** [comment]
No, but I can not imagine that you have broken things either.

**Comment by kaikreuzer** [comment]
@kgoderis Does that mean you are fine with merging this PR?

File: addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/handler/ZonePlayerHandler.java [hunk 1]
```diff
@@ -1763,6 +1774,58 @@ public void playRadio(Command command) {
 		}
 
 	}
+	
+	/**
+	 * This will attempt to match the station string with a entry in the
+	 * favorites list, this supports both single entries and playlists
+	 * 
+	 * @param favorite to match
+	 * @return true if a match was found and played.
```
Comment (evansj) [inline addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/handler/ZonePlayerHandler.java]: It is a void function...

Comment (digitaldan) [inline addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/handler/ZonePlayerHandler.java]: Good catch, obviously nothing is looking to see if this was successfull or not.

File: addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java [hunk 2]
```diff
@@ -44,7 +44,7 @@
 					"<dc:title>{2}</dc:title>" +
 					"<upnp:class>{3}</upnp:class>" +
 					"<desc id=\"cdudn\" nameSpace=\"urn:schemas-rinconnetworks-com:metadata-1-0/\">" +
-					"RINCON_AssociatedZPUDN</desc>" +
+					"{4}</desc>" +
```
Comment (kgoderis) [inline addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java]: Dan, are you sure this change does not break other stuff? 

From what I saw on the network via WireShark, that field was always set to RINCON_AssociatedZPUDN

Comment (digitaldan) [inline addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java]: Its a good question, look at line 786, by default it still uses that value unless we receive a meta data packet (its been a little while so I'm trying to remember again).  This seems to happen when playing subscription services that have login information associated with them when stored as a favorite.  I run Spotify, Google, Pandora and a I heart Radio stations bookmarked under SONOS  favorites and this works, I have not tried other services.  If we do not get this special meta data packet then it continues to use  RINCON_AssociatedZPUDN.

Comment (kgoderis) [inline addons/binding/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java]: :-) If Spotify works, then I will be a happy man. 

Initially, in the haydays, I did not have the time to figure out it had to be done.

**Comment by maggu2810** [comment]
This PR introduced multiple trailing whitespaces.

Could we use the ESH formatter, clean up, save actions rules for OH2, too?

**Comment by kaikreuzer** [comment]
I think the ESH formatter is in place already, but not everybody is yet using the new IDE setup. As long as we did not do a full re-formatting of the code, I did not want to enforce it yet.

**Pull request merged**